apps/www/registry/default/ui/multi-select.tsx

```diff
@@ -0,0 +1,161 @@
+        <PopoverTrigger asChild className={className}>
+          <Button
+            ref={ref}
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            className={`group w-full justify-between ${
+              selected.length > 1 ? "h-full" : "h-10"
+            }`}
+            onClick={() => setOpen(!open)}
+          >
+            <div className="flex flex-wrap items-center gap-1">
+              {selected.map((item) => (
+                <Badge
+                  variant="outline"
+                  key={item.value}
+                  className="flex items-center gap-1 group-hover:bg-background"
+                >
+                  {item.label}
+                  {open && (
+                    <Button
+                      asChild
+                      variant="outline"
+                      size="icon"
+                      className={`border-none duration-300 ${
+                        open ? "opacity-100 ease-in" : "opacity-0 ease-out"
+                      }`}
+                      onKeyDown={(e) => {
+                        if (e.key === "Enter") {
+                          handleUnselect(item)
+                        }
+                      }}
+                      onMouseDown={(e) => {
+                        e.preventDefault()
+                        e.stopPropagation()
+                      }}
+                      onClick={(e) => {
+                        e.preventDefault()
+                        e.stopPropagation()
+                        handleUnselect(item)
+                      }}
+                    >
+                      <X className="h-3 w-3 text-muted-foreground hover:text-foreground" />
+                    </Button>
+                  )}
+                </Badge>
+              ))}
+              {selected.length === 0 && (
+                <span>{props.placeholder ?? "Select ..."}</span>
+              )}
+            </div>
+            <ChevronsUpDown className="h-4 w-4 shrink-0 opacity-50" />
+          </Button>
+        </PopoverTrigger>
+        <PopoverContent className="w-full p-0">
+          <Command className={className}>
+            <CommandInput
+              onValueChange={(item) => {
+                console.log("dd", item)
+                setQuery(item)
+              }}
+              placeholder="Search ..."
+            />
+            <CommandEmpty>No item found.</CommandEmpty>
+            <CommandGroup className="max-h-64 overflow-auto">
+              {options.map((option) => (
+                <CommandItem
+                  key={option.value}
+                  onSelect={() => {
+                    onChange(
+                      selected.some((item) => item.value === option.value)
+                        ? selected.filter((item) => item.value !== option.value)
+                        : [...selected, option]
+                    )
+                    setOpen(true)
```

Comment:
set it to open?
if you are clicking it, the popover is already open and visible, am i missing something?